### DISCUSSION TITLE
JSDialog实现

### DISCUSSION THREAD

**Discussion by L-Super** [CefView/QCefView#277] [Q&A]
在CefViewCore里有看到相关[JSDialog](https://github.com/CefView/CefViewCore/blob/main/src/CefView/CefBrowserApp/CefViewBrowserClient_JSDialogHandler.cpp)代码，但QCefView里并未找到对应的代码，是还未实现吗？

**Comment by tishion** [comment]
是的，没有导出相关能力，你可以尝试添加

**Reply by L-Super** [reply]
> 是的，没有导出相关能力，你可以尝试添加

我想接管网页的上传功能弹出的文件对话框，即类似于奶牛快传的在线传输网站，点击上传按钮时，默认会弹出系统的文件对话框。刚刚实现了一下JSDialog，结果流程不是走这里...大佬可否指点一下，我如何接管弹出的文件对话框？

**Reply by tishion** [reply]
https://magpcss.org/ceforum/apidocs3/projects/(default)/CefDialogHandler.html

**Reply by L-Super** [reply]
> https://magpcss.org/ceforum/apidocs3/projects/(default)/CefDialogHandler.html

嗯嗯，感谢大佬，结果是我找错地方了。如果通过Qt实现了FileDialog的话，可需要PR？

**Reply by tishion** [reply]
你如果要加，应该是把这个自定义dialog的能力提供出来，而不是直接用Qt实现

**Reply by L-Super** [reply]
> 你如果要加，应该是把这个自定义dialog的能力提供出来，而不是直接用Qt实现

我实现了一版，但是不太完善，没能将dialog接口提供出来自定义，过滤类型的处理没能和浏览器的一致。望大佬指点指点
[file-dialog](https://github.com/L-Super/QCefView/commit/07109df39fb36f4265b6b1d3b8b323ef4f914ab2)

**Reply by AnsunHo** [reply]
能直接使用html表单input type="file"的吗？
